src/components/ReservationForm.tsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import './ReservationForm.css'
+import Box from '@mui/material/Box';
+import TextField from '@mui/material/TextField';
+import { MDBBtn } from 'mdb-react-ui-kit';
+import { LocalizationProvider, DatePicker } from '@mui/x-date-pickers';
+import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
+import { Col, Container, Row } from 'react-bootstrap';
+
+
+const ReservationForm = () => {
+
+    const [date1, setDate1] = React.useState<Date | null>(new Date());
+    const [date2, setDate2] = React.useState<Date | null>(new Date());
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        console.log("Fetch updated filter");
+    };
```

Comment:
Add state to the form, keep the state in an object and rename the variables to more meaningful names.
Add `onSubmit` handler to print the form values

Reply:
https://medium.com/@sandupa.egodage/react-form-with-typescript-c74510b2f9d3
Here's an article that regards this